src/controller/python/build-chip-wheel.py

```diff
@@ -0,0 +1,210 @@
+# for the installable package.
+tmpDir = os.path.join(buildDir, 'chip-wheel-components')
+
+try:
+
+    #
+    # Perform a series of setup steps prior to creating the chip package...
+    #
+
+    # Create the temporary components directory.
+    if os.path.isdir(tmpDir):
+        shutil.rmtree(tmpDir)
+    os.mkdir(tmpDir)
+
+    # Switch to the temporary directory. (Foolishly, setuptools relies on the current directory
+    # for many of its features.)
+    os.chdir(tmpDir)
+
+    # Make a copy of the chip package in the tmp directory and ensure the copied
+    # directory is writable.
+    chipPackageDir = os.path.join(tmpDir, packageName)
+    if os.path.isdir(chipPackageDir):
+        shutil.rmtree(chipPackageDir)
+    shutil.copytree(os.path.join(buildDir, 'chip'), chipPackageDir)
+    os.chmod(chipPackageDir, os.stat(chipPackageDir).st_mode|stat.S_IWUSR)
+    
+    # Copy the chip wrapper DLL from where libtool places it (.libs) into
+    # the root of the chip package directory.  This is necessary because
+    # setuptools will only add package data files that are relative to the
+    # associated package source root.
+    shutil.copy2(os.path.join(buildDir, 'chip', chipDLLName),
+                 os.path.join(chipPackageDir, chipDLLName))
+    
+    # Make a copy of the Chip Device Controller Shell script in the tmp directory,
+    # but without the .py suffix. This is how we want it to appear when installed
+    # by the wheel.
+    shutil.copy2(os.path.join(buildDir, deviceManagerShellName),
+                 os.path.join(tmpDir, chipControllerShellInstalledName))
+    
+    # Search for the CHIP LICENSE file in the parents of the source
+    # directory and make a copy of the file called LICENSE.txt in the tmp
+    # directory.  
+    def _AllDirsToRoot(dir):
+        dir = os.path.abspath(dir)
+        while True:
+            yield dir
+            parent = os.path.dirname(dir)
+            if parent == '' or parent == dir:
+                break
+            dir = parent
+    for dir in _AllDirsToRoot(buildDir):
+        licFileName = os.path.join(dir, 'LICENSE')
+        if os.path.isfile(licFileName):
+            shutil.copy2(licFileName,
+                         os.path.join(tmpDir, 'LICENSE.txt'))
+            break
+    else:
+        raise FileNotFoundError('Unable to find CHIP LICENSE file')
+    
+    # Define a custom version of the bdist_wheel command that configures the
+    # resultant wheel as platform-specific (i.e. not "pure"). 
+    class bdist_wheel_override(bdist_wheel):
+        def finalize_options(self):
+            bdist_wheel.finalize_options(self)
+            self.root_is_pure = False
+    
+    # Construct the package version string.  If building under CHIP use the CHIP
+    # build number as the package version.  Otherwise use a dummy version of '0.0'.
+    # (See PEP-440 for the syntax rules for python package versions).
+    if 'CHIP_BUILD_NUMBER' in os.environ:
```

Comment:
Please use command line arguments, not environment variables. This will allow rebuilds to happen automatically when the values change.

Reply:
Done, thanks